threatgrid/ctia
# Redefine default search fields

> **Epic** #
> Close https://github.com/threatgrid/iroh/issues/5151, https://github.com/threatgrid/iroh/issues/5269
> Related #

## Defines default fields for search.

Default fields for the full-text search are now all Entity fields with the exclusion of certain fields, like `id,` `type,` `language`, `tlp`, etc.

Users may choose to include or exclude fields as they see fit.

Basically, now when the search query is posted (and only `:query` parameter of the search provided), the search is performed in all the main (important) fields of the Entity, but excludes some second-class fields.

However, it is still possible to search in fields like `tlp` and `language` (and others) by explicitly including them in `:search_fields` parameter.


![ezgif com-gif-maker](https://user-images.githubusercontent.com/1091022/118565816-78f05500-b738-11eb-9b98-a58400eb410f.gif)


<a name="qa">[§](#qa)</a> QA
============================

### Test scenario #1

1. Create one or multiple Entities. [e.g., Incidents] with random values.
2. Set the `title` of one of the Incidents to be of some value, e.g.: `Fusce suscipit wisi nec facilisis`
2. Send a search request 
```
/ctia/incident/search?query=*suscipit*
```
3. **Expected**: It should find the Incident

### Test scenario #2

1. Create one or multiple Entities. [e.g., Incidents] with random values.
2. Set the `language` of one of the Incidents to be of some value, e.g.: `Esperanto`
2. Send a search request 
```
/ctia/incident/search?query=*Esperanto*
```
3. **Expected**: Search should yield no results, since `language` is one of the fields that are ignored by default
4. Send another search request, this time explicitly providing the field
```
/ctia/incident/search?query=*Esperanto*&search_fields=language
```
5. **Expected**: It should return the Incident


<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================
